controller/jobs.go

```diff
@@ -287,32 +266,30 @@ func (c *controllerAPI) JobLog(ctx context.Context, w http.ResponseWriter, req *
 		defer attachClient.Close()
 	}
 
-	useSSE := strings.Contains(req.Header.Get("Accept"), "text/event-stream")
-	if useSSE {
-		w.Header().Set("Content-Type", "text/event-stream; charset=utf-8")
-	} else {
-		w.Header().Set("Content-Type", "application/vnd.flynn.attach")
-	}
-	w.WriteHeader(200)
-	// Send headers right away if tailing
-	if wf, ok := w.(http.Flusher); ok && tail {
-		wf.Flush()
-	}
-
-	fw := httphelper.FlushWriter{Writer: w, Enabled: tail}
-	if useSSE {
-		ssew := NewSSELogWriter(fw)
-		exit, err := attachClient.Receive(ssew.Stream("stdout"), ssew.Stream("stderr"))
+	if strings.Contains(req.Header.Get("Accept"), "text/event-stream") {
+		ch := make(chan *sseLogChunk)
+		s := sse.NewStream(w, ch)
+		defer s.Close()
+		s.Serve()
+		exit, err := attachClient.Receive(&sseLogStream{Name: "stdout", Chan: ch}, &sseLogStream{Name: "stderr", Chan: ch})
 		if err != nil {
-			fmt.Fprintf(fw, "event: error\ndata: %s\n\n", err)
+			ch <- &sseLogChunk{Event: "error", Data: []byte(fmt.Sprintf(`"%s"`, err.Error()))}
```

Comment:
This method of serializing the error is unsafe, there could be special characters.